application/libraries/Minify.php

```diff
@@ -701,7 +701,10 @@ private function _concat_files($file_array, $directory, $out_file)
 				// if this is javascript file, check if we have ; at the end
 				if (preg_match("/.js$/i", $out_file)) {
 					if (substr(rtrim($contents), -1) !== ';') {
-						$contents .= ';';
+						$contents .= ";\n";
+					}
+					elseif (substr($contents, -1) !== "\n") {
```

Comment:
Rather than searching for the newline sign at the end of the file, we should just add `\n` between each js file.
It will be simpler and one check less.

This new condition `elseif` don't guarantee that there is no newline earlier.  So we can just always add:
```php
$contents .= "\n";
```